### PR TITLE
feat(query): support unset settings

### LINKED ISSUE #8099: Feature:  remove settings from meta
**Summary**
```
set global  xx=default;
```

this command removes xx settings from the meta-server. 

Resolve change settings name for upgrade:
```
mysql -u root -P 3307 -h 127.0.0.1 -p
Enter password:
ERROR 1815 (HY000): Unknown variable: "max_concurrent_prune"
```

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Syntax:
    
    UNSET <var> | ( <var> [, <var> ...])
    
    UNSET will remove settings keys both in meta & context.

Closes #8099

### CODE REVIEW

**Comment by BohuTANG** [comment]
Will we support: `UNSET ( <var> [ , <var> ... ] )`?

**Comment by TCeason** [comment]
> Will we support: `UNSET ( <var> [ , <var> ... ] )`?

hmm, how about this ? 

`UNSET (ALL_SETTINGS | <var> [, <var> ...])`

**Comment by sundy-li** [comment]
```
unset settings
```

Will remove settings keys both in meta & context, it's not equal to `set global setting_name = default_value;`.

**Comment by TCeason** [comment]
> ```
> unset settings
> ```
> 
> Will remove settings keys both in meta & context, it's not equal to `set global setting_name = default_value;`.

Got, before I thought use sf style was the same as [MySQL](https://github.com/datafuselabs/databend/issues/8099#issuecomment-1273931700)

File: src/query/service/src/interpreters/interpreter_unsetting.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                    (true, default_val)
+                }
+            };
+            if ok {
+                // reset the system.settings
+                self.ctx
+                    .get_settings()
+                    .set_settings(var.clone(), value.clone(), false)?;
+                // reset the ctx
+                self.ctx.set_affect(QueryAffect::ChangeSetting {
```
[the remark this answers is not in the document]
Comment (youngsofun) [reply src/query/service/src/interpreters/interpreter_unsetting.rs:68]: @Chasen-Zhang  @hantmac  does client using the affect field when running sql ` set/unset ...`

```
{
"affect":{"type":"ChangeSetting","key":"timezone","value":"UTC","is_global":false},"
}
```

[the remark this answers is not in the document]
Comment (youngsofun) [reply src/query/service/src/interpreters/interpreter_unsetting.rs:68]: if it is being used, we can change to this first:

```
{ "affect":{"type":"ChangeSetting",
   "key":"timezone","value":"UTC","is_global":false}," 
   "keys": [ ... ], 
   "values": [.. ]. 
}

```

[the remark this answers is not in the document]
Comment (youngsofun) [reply src/query/service/src/interpreters/interpreter_unsetting.rs:68]: cc @TCeason

[the remark this answers is not in the document]
Comment (hantmac) [reply src/query/service/src/interpreters/interpreter_unsetting.rs:68]: > @Chasen-Zhang @hantmac does client using the affect field when running sql ` set/unset ...`
> 
> ```
> {
> "affect":{"type":"ChangeSetting","key":"timezone","value":"UTC","is_global":false},"
> }
> ```

Thanks for notifying me,  I didn't use it.

**Review by sundy-li — APPROVED** [review]

**Comment by TCeason** [comment]
Like max_threads, now it has a default value: 16.

In connection A:

If `set (global)? max_threads=200`; 

the value of max_threads will be set to 200.

Now, we can execute `unset max_threads`  to set the value to 16. And show settings, it's 16.

But there has a issue:

the max_threads's UserSetting is 0. If connection A disconnect, and reconnect a new connection B,

the value of max_threads will be determined automatically.

I think this behavior is reasonable. cc @sundy-li @BohuTANG What do you think about it?

> BTW, the settings max_memory_usage may have the same issue.
> 
> But I try to test it, the value of max_memory_usage will be set to 0. But why databend-query can still work?
> 
> ```sql
> mysql> unset max_memory_usage;
> Query OK, 0 rows affected (0.02 sec)
> 
> mysql> show settings like 'max_memory_usage';
> +------------------+-------+---------+---------+------------------------------------------------------------------------------------------------------------------------------+--------+
> | name             | value | default | level   | description                                                                                                                  | type   |
> +------------------+-------+---------+---------+------------------------------------------------------------------------------------------------------------------------------+--------+
> | max_memory_usage | 0     | 0       | SESSION | The maximum memory usage for processing single query, in bytes. By default the value is 0 it means determined automatically. | UInt64 |
> +------------------+-------+---------+---------+------------------------------------------------------------------------------------------------------------------------------+--------+
> 
> ```

**Review by BohuTANG — APPROVED** [review]

**Comment by BohuTANG** [comment]
@TCeason For max_threads or max_memory_usage, I think we should auto detect again when unset and set the detect value.

**Comment by sundy-li** [comment]
> the max_threads's UserSetting is 0. If connection A disconnect, and reconnect a new connection B,

I think the default value of `max_threads` can't be 0, it's default value will determined automatically.

**Comment by TCeason** [comment]
> > the max_threads's UserSetting is 0. If connection A disconnect, and reconnect a new connection B,
> 
> I think the default value of `max_threads` can't be 0, it's default value will determined automatically.

Agree with you. And modify in this issue.

**Comment by BohuTANG** [comment]
@soyeric128 `UNSET`  for documentation :)

**Comment by TCeason** [comment]
> @soyeric128 `UNSET` for documentation :)

If has any questions, please notify me :)

File: src/query/settings/src/lib.rs [hunk 2]
```diff
@@ -835,6 +844,25 @@ impl Settings {
...
+
+        let tenant = self.tenant.clone();
+        let key = key.to_string();
+        let drop_handle = GlobalIORuntime::instance().spawn(async move {
+            UserApiProvider::instance()
+                .get_setting_api_client(&tenant)?
+                .drop_setting(key.as_str(), None)
+                .await
+        });
+        futures::executor::block_on(drop_handle).unwrap()?;
```
[the remark this answers is not in the document]
Comment (BohuTANG) [reply src/query/settings/src/lib.rs:861]: I think the `block_on` may cause the runtime error sometimes cc @TCeason

**Pull request merged**